fix(curriculum): prevent hardcoding pokemon (#54822)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-pokemon-search-app-project/build-a-pokemon-search-app.md

    const secondLetter = letters.charAt(Math.floor(Math.random() * charactersLength));
    const thirdLetter = letters.charAt(Math.floor(Math.random() * charactersLength));
    const fourthLetter = letters.charAt(Math.floor(Math.random() * charactersLength));
    const randomNumber1 = numbers.charAt(Math.floor(Math.random() * numbersLength));
    const randomNumber2 = numbers.charAt(Math.floor(Math.random() * numbersLength));
    
    const badName = firstLetter + secondLetter + thirdLetter + fourthLetter + randomNumber1 + randomNumber2; 

    const randomInvalidPokeId = badName; 
    searchInput.value = randomInvalidPokeId;
    searchButton.click();

    const res = await fetch('https://pokeapi-proxy.freecodecamp.rocks/api/pokemon/' + randomInvalidPokeId.toString()); // Fetch from proxy to simulate network delay

    if (!res.ok) {
      await new Promise(resolve => setTimeout(resolve, 1000)); // Additional delay to allow the alert to trigger

      assert.include(['pokémon not found', 'pokemon not found'], alertMessage.trim().replace(/[.,?!]+$/g, '').toLowerCase());
    }
  } catch (err) {
    throw new Error(err);
  }
};
```

> When the `#search-input` element contains an valid Pokemon id and the `#search-button` element is clicked, an alert should appear with the correct type.

Reviewer: ```suggestion
When the `#search-input` element contains a valid Pokemon id and the `#search-button` element is clicked, the UI should be filled with the correct data.
```
